alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/model/action/Throttle.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.alerting.model.action
+
+import org.apache.commons.codec.binary.StringUtils
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.common.xcontent.ToXContentObject
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParserUtils
+import java.io.IOException
+import java.lang.IllegalStateException
+import java.time.temporal.ChronoUnit
+import java.util.Locale
+
+data class Throttle(
+    val value: Int?,
+    val unit: ChronoUnit = ChronoUnit.MINUTES
+) : ToXContentObject {
+
+    override fun toXContent(builder: XContentBuilder, params: ToXContent.Params): XContentBuilder {
+        return builder.startObject()
+                .field(VALUE_FIELD, value)
+                .field(UNIT_FIELD, unit.name)
+                .endObject()
+    }
+
+    companion object {
+        const val VALUE_FIELD = "value"
+        const val UNIT_FIELD = "unit"
+
+        @JvmStatic
+        @Throws(IOException::class)
+        fun parse(xcp: XContentParser): Throttle {
+            var value: Int? = null
+            var unit: ChronoUnit? = null
+
+            XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, xcp.currentToken(), xcp::getTokenLocation)
+            while (xcp.nextToken() != XContentParser.Token.END_OBJECT) {
+                val fieldName = xcp.currentName()
+                xcp.nextToken()
+                when (fieldName) {
+                    UNIT_FIELD -> {
+                        val unitString = xcp.text().toUpperCase(Locale.ROOT)
+                        require(StringUtils.equals(unitString, ChronoUnit.MINUTES.name), { "Only support MINUTES throttle unit" })
```

Comment:
Why are we only supporting MINUTES for initial release of this feature?
Seems like unnecessary burden on users to do conversion to minutes when dealing with monitors with longer frequencies.

Reply:
As the throttle is just to avoid too many notifications in a short time, so we support MINUTES first. It does bring extra to do conversion to minutes. We have plan to support more units based on user's feedback.

Reply:
Is it a big change to support more units other than just MINUTES? If not I don't see why we are not supporting something like MINUTES, HOURS, DAYS from the start rather than waiting for a user to open an issue asking for us to support it. It looks like a single line change right here?

Reply:
From offline discussion: While not a big change for backend, it requires more significant UI change. Will release with this to start and iteratively increase to more units later.